packages/web/src/components/collection/desktop/CollectionActionButtons.tsx

```diff
@@ -43,6 +45,13 @@ export const CollectionActionButtons = (props: CollectionActionButtonProps) => {
     isPremium
   } = props
 
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const { data: collection } = useGetPlaylistById({
+    playlistId: typeof collectionId === 'number' ? collectionId : -1,
+    currentUserId
```

Comment:
you can use the `disabled` hook option for this:

```
const { data: collection } = useGetPlaylistById({
  playlistId: collectionId
},
{
  disabled: !collectionId  // or similar
}
```

Reply:
oh thanks! but i don't think that'll fix the types. but good to prevent the req.

Reply:
yeah you can do `playlistId: collectionId!` if it helps

Reply:
oh it's the type narrowing you need. Hmm idk prbly just have to cast it or do what you have here